python/ray/data/tests/test_stats.py

```diff
@@ -1288,6 +1300,40 @@ def test_stats_actor_cap_num_stats(ray_start_cluster):
     assert ray.get(actor._get_stats_dict_size.remote()) == (3, 2, 2)
 
 
+def test_spilled_stats(shutdown_only):
+    # The object store is about 100MB.
+    ray.init(object_store_memory=100e6)
+    # The size of dataset is 1000*(80*80*4)*8B, about 200MB.
+    ds = (
+        ray.data.range_tensor(1000, shape=(80, 80, 4), parallelism=100)
+        .map_batches(lambda x: x)
+        .materialize()
+    )
+
+    assert (
+        canonicalize(ds.stats())
+        == f"""Stage N ReadRange->MapBatches(<lambda>): N/N blocks executed in T
+* Remote wall time: T min, T max, T mean, T total
+* Remote cpu time: T min, T max, T mean, T total
+* Peak heap memory usage (MiB): N min, N max, N mean
+* Output num rows: N min, N max, N mean, N total
+* Output size bytes: N min, N max, N mean, N total
+* Tasks per node: N min, N max, N mean; N nodes used
+* Extra metrics: {MEM_SPILLED_EXTRA_METRICS}
+
+Cluster memory:
+* Spilled to disk: M
+* Restored from disk: M
+
+Dataset memory:
+* Spilled to disk: M
+"""
+    )
+
+    # Around 100MB should be spilled (200MB - 100MB)
+    assert ds._plan.stats().dataset_bytes_spilled > 100e6
```

Comment:
nice, can we also add a case with multiple Map stages which don't get fused? for example, ds.map_batches().sort().map_batches()